resources/type-system/inference.md

```diff
@@ -324,7 +329,8 @@ schema.
     `e`, using the local type inference algorithm described below with typing
     context `K`, and update `T` to be `UP(flatten(S), T)` if the enclosing
     function is `async`, or `UP(S, T)` otherwise.
-  - For each `return;` statement in the block, update `T` to be `UP(Null, T)`.
+  - If the enclosing function is not marked `sync*` or `async*`: For each
+    `return;` statement in the block, update `T` to be `UP(Null, T)`.
```

Comment:
Not: "in the block" is imprecise. It can be read to include the bodies of nested functions, and it can be read as only applying to immediate sub-statements of the block. 
Maybe "`return;` statements returning from this function" or something?

Reply:
Right, done!